backend/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -1165,6 +1165,8 @@ def get_file(self, path, local_base=None):
         try:
             try:
                 temp_file = ""
+                if not "://" in path:
+                    path = self.url + path
```

Comment:
To avoid issues with repositories urls that might contain that `://` string as part of the URL, and also to not conflict with  possible authtoken included in URL, I think it's safer to use `urlparse` and tailor the URL properly  as we already do in some other places in the `yum_src`.

Something like:

```python
try:
    if not urlparse(path).schema:    
        (s,b,p,q,f,o) = urlparse(self.url)
        if p[-1] != '/':
            p = p + '/'
        p = p + path
        url = urlunparse((s,b,p,q,f,o))
except (ValueError, IndexError, KeyError) as e:
    return None
```

And I've just also noticed the proxy parameters are not passed to the `urlgrabber` calls here: https://github.com/uyuni-project/uyuni/blob/0786975aa2b54da7642318a4c0561cdcb9ac61ee/backend/satellite_tools/repo_plugins/yum_src.py#L1178 and neither here: https://github.com/uyuni-project/uyuni/blob/0786975aa2b54da7642318a4c0561cdcb9ac61ee/backend/satellite_tools/repo_plugins/yum_src.py#L1182

So, in order to incorporate the correct proxy parameters to the two urlgrabber calls, we should do something like:

```python
        urlgrabber_opts = {}
        self.set_download_parameters(urlgrabber_opts, path, temp_file)
        downloaded = urlgrabber.urlgrab(path, temp_file, **urlgrabber_opts)
        ...
        urlgrabber_opts = {}
        self.set_download_parameters(urlgrabber_opts, path)
        return urlgrabber.urlread(path, **urlgrabber_opts)
```

Notice that `self.set_download_parameters` needs to be adapted to allow passing no `target_file` and simply do not return this attribute in the parameters.

Reply:
Gold plating :P
1. Tested and adapted your suggested URL detection. Took the exception untested as granted.
2. I did not actually test the proxy changes directly but verified them and they look ok. Basic kickstart file download (still) works. (and so does a reposync...)